d3ploy/demand_driven_deployment_inst.py

```diff
@@ -277,18 +310,23 @@ def calc_diff(self, commod, time):
         return diff, supply, demand
 
     def predict_supply(self, commod):
+        if self.installed_cap:
+            input = self.installed_capacity[commod]
+        else:
+            input = self.commodity_supply[commod]
+
         if self.calc_method in ['arma', 'ma', 'arch']:
-            supply = CALC_METHODS[self.calc_method](self.commodity_supply[commod],
+            supply = CALC_METHODS[self.calc_method](input,
```

Comment:
here, just feed it `target()` from the function suggested above.

Reply:
I implemented this